File: src/test/java/com/baulsupp/oksocial/services/twitter/TwurlCredentialsStoreTest.java
package com.baulsupp.oksocial.services.twitter;

import com.baulsupp.oksocial.services.twitter.twurlrc.TwurlCredentialsStore;
import java.io.File;
import java.io.IOException;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class TwurlCredentialsStoreTest {
  @Test public void testReadDefaultCredentials() throws IOException {
    File file =
        new File(TwurlCredentialsStoreTest.class.getResource("/single_twurlrc.yaml").getFile());
    TwurlCredentialsStore store = new TwurlCredentialsStore(file);

    TwitterCredentials credentials = store.readCredentials().get();

    assertEquals("PROFILE", credentials.username);
    assertEquals("CONSUMER_KEY", credentials.consumerKey);
    assertEquals("CONSUMER_SECRET", credentials.consumerSecret);
    assertEquals("1234-TOKEN", credentials.token);
    assertEquals("SECRET", credentials.secret);
  }
}

File: src/test/kotlin/com/baulsupp/oksocial/services/twitter/TwurlCredentialsStoreTest.kt
package com.baulsupp.oksocial.services.twitter

import com.baulsupp.oksocial.services.twitter.twurlrc.TwurlCredentialsStore
import org.junit.Test


import java.io.File
import java.io.IOException
import kotlin.test.assertEquals

class TwurlCredentialsStoreTest {
    @Test
    @Throws(IOException::class)
    fun testReadDefaultCredentials() {
        val file = File(TwurlCredentialsStoreTest::class.java.getResource("/single_twurlrc.yaml").file)
        val store = TwurlCredentialsStore(file)

        val credentials = store.readCredentials()!!

        assertEquals("PROFILE", credentials.username)
        assertEquals("CONSUMER_KEY", credentials.consumerKey)
        assertEquals("CONSUMER_SECRET", credentials.consumerSecret)
        assertEquals("1234-TOKEN", credentials.token)
        assertEquals("SECRET", credentials.secret)
    }
}
